### ISSUE TITLE
Phase09-Part1-AmirMahdi Kousheshi

### ISSUE THREAD

**Issue by amkamir82** [Star-Academy/codestar-intern-issues#181]
This issue template is based of [Phase09-Web.md](https://github.com/Star-Academy/codestar-internship/blob/master/Projects/Phase09-Web/Phase09-Web.md) contents.
Click on "Raw" button on right side of this page to see original markdown then create a new issue with orginal content of this tasklist:

# Phase 9 First Part

- [x] Before starting: Read the link about history and purpose of creating & developing Web, then answer this questions:
  - [x] How did developers provide server-client softwares for users before creation of web technologies?
  
      `[they maybe run their back end of their sites in some servers and it does not any front end and some thing like that same as now]`
      
  - [x] What was the main trouble with traditional technologies before web?
  
      `[I think they could not edit their web pages as simple as now]`
      
  
  - [x] How did web overcome this main trouble?
  
      `[with some applications like javascripts and ETC it could overcame the troubles]`

## Asp.net core

- [x] Getting Started with HTTP:
  - [x] What does a protocol mean?
  
    `[some rules standings together provide a protocol]`

  - [x] Why designing a protocol for communication is important prerequirement of all other web related technologies?
  
    `[because they should follow same rules to communicate to each others ]`
    
  - [x] Try to get familier with HTTP by reading links.
  - [x] Why HTTP is most popular protocol in web technologies?
    
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
    
  - [x] What is the most interesting aspect of HTTP in your opinion? 
  
    `[HTTP is simple]`

- [ ] Getting Started with REST:
  - [ ] Read mentioned links.
  - [ ] Why do we need RESTful APIs?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`

- [ ] Getting Started with Asp.net Core:
  - [ ] Read mentioned links.
  - [ ] Create a hello world API.
  - [ ] Call your api from browser.
  - [ ] Play with routing options! (Try to use it & enjoy it!)
  - [ ] Why do we need configurable routings in web development?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
  
  - [ ] Play with dependency injection! (Define an interface & inject interface implementation to code)
  - [ ] Why do we need dependency injection?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
  - [ ] Play with parameter binding! (use different modes)
  - [ ] Install postman.
  - [ ] Call your APIs with postman.
  
- [ ] Implement Asp.net Core Project:
  - [ ] Design your search API.
  - [ ] Implement & Test your API.

# Review
Link of your own PR:

`[FILL HERE WITH LINK OF YOUR PR.]`

Link of other PR(s) you agreed to review:

`[FILL HERE WITH LINK OF PR(s) YOU REVIEWED.]`

- [ ] Your own PR is reviewed and approved by at least one other team.

- [ ] Your own PR is reviewed and approved by mentor (hamyar).

- [ ] Your own PR is reviewed and approved by manager (rahbar).

- [ ] Your own PR is merged.
